### ISSUE TITLE
Reactlog performance regression with large objects

### ISSUE THREAD

**Issue by wch** [rstudio/shiny#2375]
When an application sets a `reactiveValues` to a value that is a large list that takes a long time to print with `str()`, it can slow down the shiny application even when reactlog is disabled.

This is primarily an issue when the value is a nested list. With the value is a large flat list, `str()` by default limits the output to 99 items. However, when it's a nested list, each sublist prints up to 99 items, so the final output can be quite large, and quite slow.

Example:

```R
obj <- lapply(1:60, function(i) {
  lapply(1:60, function(j) {
    list(i, j)
  })
})

# Check how long it takes to call `str()` on the object
system.time({
  capture.output(str(obj))
})
#>   user  system elapsed 
#>  4.006   0.069   4.093 


library(profvis)
library(shiny)


profvis({
  runApp(
    shinyApp(
      
      ui = fluidPage(
        textOutput("txt")
      ), 
      
      server = function(input, output) {
        r <- reactiveValues(x = 1)
        
        observe({
          r$x <- obj
        })
        
        output$txt <- renderText({
          length(r$x)
        })
      }
    )
  )
})
```

This takes about 6 seconds to display anything in the Shiny app. In Shiny 1.2.0, it is almost instantaneous.

The profile indicates that most of the time is spent in `rLog$defineKey`. (Unchecking "Hide internal function calls" in the profile shows useful information.)

![image](https://user-images.githubusercontent.com/86978/55892762-89f33300-5b7c-11e9-812e-da7ac9ac3b36.png)

**Comment by wch** [comment]
For now, it's possible to work around it by redefining `str.list` inside the application, like so:

```R
str.list <- function(x, ...) ""
```

**Comment by wch** [comment]
This was fixed by #2377.
